services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/events/SignalStateEvent/SignalStateEventRepositoryImpl.java

```diff
@@ -0,0 +1,104 @@
+
+package us.dot.its.jpo.ode.api.accessors.events.SignalStateEvent;
+
+import java.util.Date;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.stereotype.Component;
+import org.springframework.data.domain.Sort;
+
+import us.dot.its.jpo.conflictmonitor.monitor.models.events.StopLinePassageEvent;
+
+import org.springframework.data.mongodb.core.aggregation.Aggregation;
+import org.springframework.data.mongodb.core.aggregation.AggregationResults;
+import org.springframework.data.mongodb.core.aggregation.DateOperators;
+
+import us.dot.its.jpo.ode.api.ConflictMonitorApiProperties;
+import us.dot.its.jpo.ode.api.models.IDCount;
+
+@Component
+public class SignalStateEventRepositoryImpl implements SignalStateEventRepository {
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Autowired
+    ConflictMonitorApiProperties props;
+
+    private final String collectionName = "CmStopLinePassageEvent";
```

Comment:
Question (non-blocking): Can you confirm if this is the correct collection name for this class? It doesn’t seem to align with the class name.

Reply:
This is correct for now, but needs to be changed. Here is the backlog task to sync the collection name with the event/class: https://dev.azure.com/SOC-OIT/CDOT/_backlogs/backlog/CDOT%20Connected%20Vehicles/Backlog%20items/?workitem=290340